hibernate4/src/main/java/com/fasterxml/jackson/datatype/hibernate4/HibernateProxySerializer.java

```diff
@@ -194,4 +200,40 @@ protected Object findProxied(HibernateProxy proxy)
         }
         return init.getImplementation();
     }
+    
+    // Alas, hibernate offers no public api to access this information, so we must resort to ugly hacks ...
```

Comment:
What I meant was just to explain what is being done; mention that there's no public API is fine, but explain what information, and for what reason. If that is covered in other parts that may be fine.
There's no need (IMO) to explain what else has been tried; just what the intent is.